src/main/java/net/openhft/chronicle/threads/EventGroup.java

```diff
@@ -386,11 +388,36 @@ protected void performStart() {
             try {
                 timeoutPauser.pause(WAIT_TO_START_MS, TimeUnit.MILLISECONDS);
             } catch (TimeoutException e) {
+                Jvm.error().on(EventGroup.class, format("Timed out waiting for start!%n" +
+                                "%s%n%n" +
+                                "%s%n%n",
+                        dumpEventLoopState("Core", core),
+                        dumpEventLoopState("Monitor", monitor)));
                 throw Jvm.rethrow(e);
             }
         }
     }
 
+    private String dumpEventLoopState(String name, @Nullable EventLoop eventLoop) {
+        if (eventLoop == null) {
+            return name + " event loop is null";
+        }
+        StringBuilder builder = new StringBuilder();
+        builder.append(name).append(" event loop state\n");
+        builder.append(eventLoop).append('\n');
+        if (eventLoop instanceof MediumEventLoop) {
+            Thread t = ((MediumEventLoop) eventLoop).thread();
+            if (t != null) {
+                builder.append("Thread state: ").append(t.getState()).append('\n');
+                builder.append("Stack trace:\n");
+                Jvm.trimStackTrace(builder, t.getStackTrace());
+            } else {
+                builder.append("Thread is null\n");
+            }
+        }
+        return builder.toString();
+    }
```

Comment:
This is a lot of junk to put into the EventGroup. I wonder if it'd be better to put it into the toString for the respective EventLoops, but it's also noisy and potentially expensive to generate.

Any recommendations for other things to put in here? I'd like to put lifecycle in there, but I'm reluctant for that to leak out of the parent class, it's an internal detail.

Reply:
Is there any point trying to look at the thread? If MediumEventLoop.isAlive is false then thread by definition will be null

Reply:
Not so... it can also be when `Thread#isAlive()` returns false.
https://github.com/OpenHFT/Chronicle-Threads/blob/b20c825ee94abcc863a1b8a7d48f8a48b3c57be0/src/main/java/net/openhft/chronicle/threads/MediumEventLoop.java#L565-L569
>  A thread is alive if it has been started and has not yet died.

I'm not sure what it will look like, but we might be able to distinguish between "not started" and "dead"

Reply:
ah yes right you are